sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql

```diff
@@ -0,0 +1,222 @@
+    first_seen.sample_id AS sample_id,
+    first_seen.first_seen_date AS first_seen_date,
+    DATE(first_seen.first_run_datetime) AS first_run_date,
+    first_seen.first_reported_country AS first_reported_country,
+    first_seen.first_reported_isp AS first_reported_isp,
+    first_seen.channel AS channel,
+    first_seen.device_manufacturer AS device_manufacturer,
+    first_seen.device_model AS device_model,
+    first_seen.os_version AS os_version,
+    first_session.adjust_campaign AS adjust_campaign,
+    first_session.adjust_ad_group AS adjust_ad_group,
+    first_session.adjust_creative AS adjust_creative,
+    COALESCE(first_session.adjust_network, metrics.adjust_network) AS adjust_network,
+    metrics.install_source AS install_source,
+    STRUCT(
+      CASE
+      WHEN
+        first_session.client_id IS NULL
+      THEN
+        FALSE
+      ELSE
+        TRUE
+      END
+      AS reported_first_session_ping,
+      CASE
+      WHEN
+        metrics.client_id IS NULL
+      THEN
+        FALSE
+      ELSE
+        TRUE
+      END
+      AS reported_metrics_ping,
+      DATE(first_session.first_run_datetime) AS min_first_session_ping_run_date,
+      CASE
+        mozfun.norm.get_earliest_value(
+          [
+            (
+              STRUCT(CAST(first_session.adjust_network AS STRING), first_session.first_run_datetime)
+            ),
+            (STRUCT(CAST(metrics.adjust_network AS STRING), metrics.min_submission_datetime))
+          ]
+        )
+      WHEN
+        STRUCT(first_session.adjust_network, first_session.first_run_datetime)
+      THEN
+        'first_session'
+      WHEN
+        STRUCT(metrics.adjust_network, metrics.min_submission_datetime)
+      THEN
+        'metrics'
+      ELSE
+        NULL
+      END
+      AS adjust_network__source_ping,
+      CASE
+      WHEN
+        metrics.install_source IS NOT NULL
+      THEN
+        'metrics'
+      ELSE
+        NULL
+      END
+      AS install_source__source_ping,
+      mozfun.norm.get_earliest_value(
+        [
+          (STRUCT(CAST(first_session.adjust_network AS STRING), first_session.first_run_datetime)),
+          (STRUCT(CAST(metrics.adjust_network AS STRING), metrics.min_submission_datetime))
+        ]
+      ).earliest_date AS adjust_network__source_ping_datetime,
+      CASE
+      WHEN
+        metrics.install_source IS NOT NULL
+      THEN
+        metrics.min_submission_datetime
+      ELSE
+        NULL
+      END
+      AS install_source__source_ping_datetime
+    ) AS metadata
+  FROM
+    first_seen
+  FULL OUTER JOIN
+    first_session_ping first_session
+  USING
+    (client_id)
+  FULL OUTER JOIN
+    metrics_ping AS metrics
+  USING
+    (client_id)
+),
+--existing clients in firefox_android_clients_v1
+_previous AS (
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.fenix_derived.firefox_android_clients_v1`
+)
+SELECT
+  IF(_current.client_id IS NOT NULL, _current, _previous).* REPLACE (
```

Comment:
Should we default to `_previous`? Not sure offhand, just want to make sure this is right

Reply:
I see it correct.

Do you mean
    `_previous.client_id IS NULL, _current, _previous` or
    `_previous.client_id IS NOT NULL, _previous, _current` ?